File: LeetCode/Easy/0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    minPrice, maxProfit := prices[0], 0
    for i:=0; i<len(prices); i++{
        
        minPrice = min(minPrice, prices[i])
        maxProfit = max(maxProfit, (prices[i]-minPrice))
    }
    return maxProfit
}

File: LeetCode/Easy/0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0, sum = 0;
        int l = 0, r = 1;
        while (r < prices.size()) {
            if (prices[r] - prices[l] < 0) {
                l = r;
            } else {
                profit = max(profit, prices[r] - prices[l]);
            }
            r++;
        }
        // profit = max(profit, sum);
        return profit;
    }
};